r/tests/testthat/test-dplyr-mutate.R

```diff
@@ -279,16 +279,16 @@ test_that("dplyr::mutate's examples", {
   # Examples we don't support should succeed
   # but warn that they're pulling data into R to do so
 
-  # across and autosplicing: ARROW-11699
   compare_dplyr_binding(
     .input %>%
       select(name, homeworld, species) %>%
       mutate(across(!name, as.factor)) %>%
       collect(),
     starwars,
-    warning = "Expression across.*not supported in Arrow"
   )
```

Comment:
This test now fails with the following error:

```
Error in `collect()`:
! Invalid: ExecuteScalarExpression cannot Execute non-scalar expression dictionary_encode(homeworld, {null_encoding_behavior=MASK})
/home/nic2/arrow/cpp/src/arrow/compute/exec/project_node.cc:91  ExecuteScalarExpression(simplified_expr, target, plan()->exec_context())
/home/nic2/arrow/cpp/src/arrow/compute/exec/exec_plan.cc:559  iterator_.Next()
/home/nic2/arrow/cpp/src/arrow/record_batch.cc:337  ReadNext(&batch)
/home/nic2/arrow/cpp/src/arrow/record_batch.cc:351  ToRecordBatches()
Run `rlang::last_error()` to see where the error occurred.
```

Reply:
There already is at least one issue for `as.factor` in Acero: ARROW-12632, and I think there was another. 

I'd change this test to use a different function that is supported.

Reply:
Though we should probably make sure that `as.factor` isn't mapped to anything in the dplyr funcs for now, so that this fails in the right way rather than try to collect.

Reply:
I've opened [a ticket](https://issues.apache.org/jira/browse/ARROW-17371) and [associated PR](https://github.com/apache/arrow/pull/13836) to remove the existing mapping.